### ISSUE TITLE
[FR] implement everything Omnidroid can

### ISSUE THREAD

**Issue by ildar** [renyuneyun/Easer#14]
OD can be found on F-Droid site

**Comment by ildar** [comment]
emm, probably [libretasks](https://f-droid.org/repository/browse/?fdid=libretasks.app) is a more fresh thing to consider.

**Comment by fazlerabbi37** [comment]
@renyuneyun any update on this?

---

maybe it would help if we could list the features of LibreTasks. I would try to do in this weekend @ildar what do you think?

**Comment by ildar** [comment]
Positive. The issue is useless without the list.

**Comment by fazlerabbi37** [comment]
As far as I understood in LibreTasks you can choose root events to trigger a rule. You can add one or more filter with the events. The filter can have one or more sibling or sub filters. Finally, you add one or more actions.

##### root event
- [ ] Batter is low
- [ ] Batter is ok after low
- [ ] Camera button was pressed
- [ ] GPS Location Changed
- [ ] Internet Available
- [ ] Missed Call
- [ ] Phone Call Ended
- [ ] Phone is Ringing
- [ ] Power Disconnected
- [ ] Power connected
- [ ] SMS Received
- [ ] Service Available
- [ ] Time is

##### filter
- [ ] Time
    - [ ] Time is (daily)
    - [ ] Time is not (daily)
    - [ ] Time is before (daily)
    - [ ] Time is not before (daily)
    - [ ] Time is after (daily)
    - [ ] Time is during (daily)
    - [ ] Time is not during (daily)
    - [ ] Time is day of week
- [ ] Location
    - [ ] Location near
    - [ ] Location away

##### action
- [ ] Media
    - [ ] Play Media
    - [ ] Pause Media
- [ ] LibreTasks
    - [ ] Show Web Site
    - [ ] Display Alert
    - [ ] Power Off Device
    - [ ] Notify Message
- [ ] Email
    - [ ] GMAIL SEND
- [ ] SMS
    - [ ] SMS Send
- [ ] Signals
    - [ ] Turn On Wifi
    - [ ] Turn Off Bletooth
    - [ ] Turn Off Wifi
    - [ ] Turn On Bletooth
- [ ] Setting
    - [ ] Set Phone Loud
    - [ ] Change screen brightness
    - [ ] Set Phone Silent
    - [ ] Set Phone Vibrate
- [ ] Phone
    - [ ] Dial Number

---

@ildar do add or suggests if I missed anything.

**Comment by renyuneyun** [comment]
@fazlerabbi37 Thank you very much for providing the list!
I ticked the boxes with the functions I believe implemented with certain. Here are some other discussions:

* Internet Available: Am I safe to say by using network communication event, you can test the availability of Internet?
* Missed Call: this should (in principle) be possible by using the phone call event in this way: ringing -> NOT ongoing -> idle?
* Service Available: I remember this refers to LibreTask's background service is started. So I guess I can say it's "implemented" (automatically)?


* Location: v0.8-pre-beta includes an implementation of geo location event & condition. (#79)

**Comment by fazlerabbi37** [comment]
This is great news! I suppose, we can do the missed call event but I would let you know for sure after testing. The location one would be great as I am waiting for it for a long time.

> Internet Available: Am I safe to say by using network communication event, you can test the availability of Internet?

Assuming you are referring to the TCP Trip event, I really haven't been able to figure that one yet. 

> Service Available: I remember this refers to LibreTask's background service is started. So I guess I can say it's "implemented" (automatically)?

I don't get the chance to know about it in details either. I am really sorry that I couldn't be of more help. Maybe @ildar can help us out?

**Comment by ildar** [comment]
This FR isn't about "I need these features now" but about "Easer could have
these". So no hurry in implementing everything from this list.
Sorry I didn't write clear enough from the beginning.

**Comment by Soundtoxin** [comment]
I just installed Easer for the first time. So, it's not yet possible to open a specific URL? I was hoping to open a page in my browser at a certain time each day. I found this issue while checking if it could be done.

**Comment by ildar** [comment]
Should be possible through intents albeit it's a hackish way.

**Comment by renyuneyun** [comment]
@Soundtoxin I agree with ildar. It's possible to do it through Intents, and I remember it's *the* way to do it (underlying-ly) if such a skill is developed.
But of course, having a skill for this will have a better UX. Unfortunately it's not developed at the moment.
